src/@batch-flask/ui/workspace/feature-visible.directive.ts

```diff
@@ -0,0 +1,39 @@
+import { Directive, Input, OnDestroy, OnInit, TemplateRef, ViewContainerRef } from "@angular/core";
+import { Subscription } from "rxjs";
+import { WorkspaceService } from "./workspace.service";
+
+@Directive({
+    selector: "[blFeatureOn]",
+})
+export class FeatureVisibleDirective implements OnInit, OnDestroy  {
+    // tslint:disable-next-line:no-input-rename
+    @Input("blFeatureOn") public feature: string;
+
+    private _subscription: Subscription;
+
+    constructor(
+        private templateRef: TemplateRef<any>,
+        private viewContainer: ViewContainerRef,
+        private workspaceService: WorkspaceService,
+    ) {
+    }
+
+    public ngOnInit() {
+        this._subscription = this.workspaceService.currentWorkspace.subscribe((workspace) => {
+            if (this.feature && workspace) {
```

Comment:
you need to listen to the `ngOnChanges` here too otherwise if feature changes then it won't update

Reply:
Why would the feature change?

Reply:
why not, might be some computed features

Reply:
Yeah I guess that's true. Future proofing is not a bad idea.